pkg/sql/plpgsql/parser/parse.go

```diff
@@ -116,18 +118,51 @@ func (p *Parser) parse(
 			err = errors.WithTelemetry(err, tkeys...)
 		}
 
-		return statements.PLPGStatement{}, err
+		return statements.PLpgStatement{}, err
 	}
-	return statements.PLPGStatement{
+	return statements.PLpgStatement{
 		AST:             p.lexer.stmt,
 		SQL:             sql,
 		NumPlaceholders: p.lexer.numPlaceholders,
 		NumAnnotations:  p.lexer.numAnnotations,
 	}, nil
 }
 
+// CountPLpgSQLStmt parses the function body and calls the Walk function for
+// each plpgsql statement.
+func CountPLpgSQLStmt(sql string) (plpgsqltree.PLpgSQLStmtCounter, error) {
+	v := plpgsqltree.MakePLpgSQLVisitor()
+	stmt, err := Parse(sql)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, s := range stmt.AST.Body {
```

Comment:
Heh, `PLpgSQLStmtBlock` should be also a statement, No? Namely, looks like you don't need to do the loop here, instead do it in the `walkStmt` of `PLpgSQLStmtBlock`?